packages/csvviewer-extension/src/searchprovider.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+import {
+  ISearchProvider,
+  ISearchMatch
+} from '@jupyterlab/documentsearch-extension';
+import { CSVViewer } from '@jupyterlab/csvviewer';
+import { IDocumentWidget, DocumentWidget } from '@jupyterlab/docregistry';
+import { Signal, ISignal } from '@phosphor/signaling';
+
+export class CSVSearchProvider implements ISearchProvider {
+  /**
+   * Report whether or not this provider has the ability to search on the given object
+   */
+  static canSearchOn(domain: any): boolean {
+    // check to see if the CMSearchProvider can search on the
+    // first cell, false indicates another editor is present
+    return (
+      domain instanceof DocumentWidget && domain.content instanceof CSVViewer
```

Comment:
The tracker won't work here because the csv viewer doesn't expose a tracker. Granted that's a problem with the csv viewer, but I was trying to limit scope here in this PR.